src/lib/Microsoft.Health.Events/EventConsumers/Service/Infrastructure/EventQueue.cs

```diff
@@ -0,0 +1,92 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Health.Events.Model;
+
+namespace Microsoft.Health.Events.EventConsumers.Service.Infrastructure
+{
+    public class EventQueue
```

Comment:
nit:EventPartition or something similar seems more appropriate since this is encapsulating a queue and related partition data.